test/timer.js

```diff
@@ -105,5 +105,32 @@ module.exports = function runTimerTestSuite() {
         });
       });
     });
+
+    it('should record "user time" of promise', function () {
+      /* globals Promise */
+      var statsd = new StatsD({mock:true});
+
+      var onehundredMsFunc = function () { return delay(100); };
+
+      var instrumented = statsd.asyncTimer(onehundredMsFunc, 'name-thingy');
+
+      return instrumented().then(function() {
+
+        var stat = statsd.mockBuffer[0];
+        var name = stat.split(/:|\|/)[0];
+        var time = stat.split(/:|\|/)[1];
+
+        assert.equal(name, 'name-thingy');
+        assert.ok(parseFloat(time) >= 100);
+        assert.ok(parseFloat(time) < 110);
```

Comment:
I have the same worry here about creating a jittery test, perhaps a little higher?